mesonbuild/minstall.py

```diff
@@ -89,6 +89,44 @@ def sanitize_permissions(path, umask):
     except PermissionError as e:
         msg = '{!r}: Unable to set permissions {!r}: {}, ignoring...'
         print(msg.format(path, new_perms, e.strerror))
+    except FileNotFoundError as e:
+        if os.path.islink(path):
+            msg = '{!r}: Broken symlink: ignoring...'
+            print(msg.format(path))
+        else:
+            raise
+
+def set_chown(path, user=None, group=None, dir_fd=None, follow_symlinks=True):
+    """Change owner user and group of the given path.
+    user and group can be the uid/gid or the user/group names, and in that case,
+    they are converted to their respective uid/gid.
+    This is a copy of shutil.chown, but support for all parameters supported by
+    os.chown
+    """
+
+    if user is None and group is None:
+        raise ValueError("user and/or group must be set")
+
+    _user = user
+    _group = group
+
+    # -1 means don't change it
+    if user is None:
+        _user = -1
+    # user can either be an int (the uid) or a string (the system username)
+    elif isinstance(user, str):
+        _user = shutil._get_uid(user)
+        if _user is None:
+            raise LookupError("no such user: {!r}".format(user))
+
+    if group is None:
+        _group = -1
+    elif not isinstance(group, int):
+        _group = shutil._get_gid(group)
+        if _group is None:
+            raise LookupError("no such group: {!r}".format(group))
+
+    os.chown(path, _user, _group, dir_fd=dir_fd, follow_symlinks=follow_symlinks)
```

Comment:
I noticed that this was [taken from CPython](https://github.com/python/cpython/blob/master/Lib/shutil.py#L1172) and frankly I am shocked that this trivial change doesn't already exist upstream. Fancy opening an upstream bug with a patch so we can eventually get rid of this? :)

Reply:
Yes, I didn't want to risk to change the behavior at all, so I just reused that.

Actually [there is](https://bugs.python.org/issue18108), and I wanted to re-propose the PR already.

Reply:
Copying code verbatim from Python (or anywhere else for that matter) is ABSOLUTELY FORBIDDEN. At least it is a license violation and at worst it is illegal (assuming the licenses are not compatible, I don't know for sure and am not a lawyer). This will not be accepted UNDER ANY CIRCUMSTANCES!

Reply:
> At least it is a license violation

For some reason I was under the impression that we're licensed under the same license as CPython.

Reply:
@jpakkane well, ok, I can rephrase that code.... Won't change much though, nor I though it was a license violation here.